test/util/bspNet/block.ts

```diff
@@ -397,16 +398,23 @@ export const advanceToBlock = async (
 
     // Check if we need to wait for BSP proofs.
     if (options.watchForBspProofs) {
+      let txsToWaitFor = 0;
       for (const challengeBlockNumber of challengeBlockNumbers) {
         if (currentBlockNumber === challengeBlockNumber.nextChallengeBlock) {
-          // Wait for the BSP to process the proof.
-          await sleep(500);
+          txsToWaitFor++;
 
           // Update next challenge block.
           challengeBlockNumbers[0].nextChallengeBlock += challengeBlockNumber.challengePeriod;
           break;
```

Comment:
Should this break be here? What's its purpose?

Reply:
No it shouldn't, you're right. There could be more than one BSP having its next challenge in `currentBlockNumber`, in which case we need to exhaustively check all the items in `challengeBlockNumbers`.